Improve Docs for Decentralized Exchange

mDuo13 opened the pull request:
- Add a new concept article describing the DEX at a high level, including a new diagram of processing a trade. (Fixes #1036)
    - The `.uxf` file is the [Umlet](https://www.umlet.com/) source file for the diagram. The `.svg` file is the exported version of the same thing.
- Rewrite the "Offers" concept page to better complement the DEX concept page and hopefully read more clearly without going immediately into technical details.
- Update the OfferCreate transaction reference slightly to add a list of error cases and improve wording a bit, including linking to the new articles.

mDuo13 committed:
Decentralized Exchange concept article

1st draft

mDuo13 committed:
DEX: example trade diagram, offers rewording

mDuo13 committed:
Offers: more rewriting/updates

mDuo13 committed:
OfferCreate: Add error cases + other updates

amarantha-k commented on content/concepts/decentralized-exchange/decentralized-exchange.md:
```diff
@@ -0,0 +1,74 @@
+---
+# Decentralized Exchange
+
+The XRP Ledger's has possibly the world's oldest _decentralized exchange_ (sometimes abbreviated "DEX"), operating continuously since the XRP Ledger's launch in 2012. The exchange allows users to buy and sell [tokens](tokens.html) for XRP or other tokens, with minimal [fees](fees.html) charged to the network itself (not paid out to any party).
```
The XRP Ledger has ...

amarantha-k commented on content/concepts/decentralized-exchange/decentralized-exchange.md:
```diff
@@ -0,0 +1,74 @@
+
+The XRP Ledger's has possibly the world's oldest _decentralized exchange_ (sometimes abbreviated "DEX"), operating continuously since the XRP Ledger's launch in 2012. The exchange allows users to buy and sell [tokens](tokens.html) for XRP or other tokens, with minimal [fees](fees.html) charged to the network itself (not paid out to any party).
+
+**Caution:** Any can [issue a token](issue-a-fungible-token.html) with any currency code or ticker symbol they want and sell it in the decentralized exchange. Always pay attention to the issuer of any token you buy tokens; otherwise, you might give up something of value and receive worthless tokens in exchange.
```
Anyone can [issue a token](issue-a-fungible-token.html) with any currency code or ticker symbol they want and sell it in the decentralized exchange. Always perform due diligence before buying a token and pay attention to the issuer of the token. Otherwise, you might give up something of value and receive worthless tokens in exchange.

amarantha-k commented on content/concepts/decentralized-exchange/decentralized-exchange.md:
```diff
@@ -0,0 +1,74 @@
+
+{{ include_svg("img/decentralized-exchange-example-trade.svg", "Diagram: Partially filled offer to buy a token for XRP.") }}
+
+The above diagram shows an example trade in the decentralized exchange. In this example, a trader named Tran places an Offer to buy 100 tokens with the currency code FOO issued by a fictional business called WayGate. (For brevity, "FOO.WayGate" refers to these tokens.) Tran specifies that he is willing to spend up to 1000 XRP for the full total. When Tran's transaction is processed, all of the fo ...
```
Replace "all of the following things happen" with "the following things happen"

amarantha-k commented on content/concepts/decentralized-exchange/decentralized-exchange.md:
```diff
@@ -0,0 +1,74 @@
+
+The XRP Ledger does not natively represent concepts such as market orders, stop orders, or trading on leverage. Some of these may be possible with creative use of custom tokens and Offer properties.
+
+As a decentralized system, the XRP Ledger does not have any information on the actual people and organizations behind the [accounts](accounts.html) involved in trading. Therefore, the ledger itself cannot implement restrictions around who can or cannot participate in trading, and users and issuers must take care to follow any relevant laws to regulate trading tokens that represent various types o ...
```
Suggestion: Change "help issuers to follow these types of rules" to "help issuers comply with relevant laws and regulations" ?

amarantha-k commented on content/concepts/decentralized-exchange/offers.md:
```diff
@@ -7,85 +7,107 @@ labels:
 
-It is possible for an offer to become temporarily or permanently _unfunded_:
+- Hold any positive amount of that token, _or_
+- be the token's issuer.
```
Be

amarantha-k commented on content/concepts/decentralized-exchange/offers.md:
```diff
@@ -7,85 +7,107 @@ labels:
+However, you don't need to hold the full amount specified in the Offer. Placing an Offer does not lock up your funds, so you can place multiple Offers to sell the same tokens (or XRP), or place an Offer and hope to get enough tokens or XRP to fully fund it later.
 
-An unfunded offer can stay on the ledger indefinitely, but it does not have any effect. The only ways an offer can be *permanently* removed from the ledger are:
+**To sell XRP,** you must hold enough XRP to meet all the [reserve requirements](reserves.html), including the reserve for the Offer object to be placed in the ledger and the trust line to hold the token you are buying. As long as you have a nonzero amount of XRP left over after setting aside the amounts for the reserves, you can place the Offer.
```
- non-zero
- "setting aside the amounts for " -> "setting aside the amount for"

amarantha-k commented on content/concepts/decentralized-exchange/offers.md:
```diff
@@ -7,85 +7,107 @@ labels:
-However, holding non-XRP balances still requires a trust line to the address issuing those balances. When an offer is taken, it automatically creates any necessary trust lines, setting their limits to 0. Because [trust lines increase the reserve an account must hold](reserves.html), any offers that would require a new trust line also require the address to have enough XRP to meet the reserve for  ...
+However, holding tokens still requires a trust line to the issuer. When an Offer is consumed, it automatically creates any necessary trust lines, setting their limits to 0. Because [trust lines increase the reserve an account must hold](reserves.html), any Offers that would require a new trust line also require the address to have enough XRP to meet the reserve for that trust line.
 
 Trust lines limits protect you from receiving more of a token as payment than you want. Offers can go beyond those limits because they are an explicit statement of how much of the token you want.
```
Trust line limits?

amarantha-k commented on content/references/protocol-reference/transactions/transaction-types/offercreate.md:
```diff
@@ -38,29 +36,40 @@ For more information about how Offers work, see [Offers](offers.html).
-
-
 
+## Error Cases
```
Suggest presenting this information as a table with error codes and description.

mDuo13 committed:
Known Amendments: fix tecEXPIRED, DepositPreauth

- tecEXPIRED was initially attached to the Checks amendment but got
  moved to DepositPreauth before either became released. The docs weren't
  previously updated to reflect this change
- A change to the snippets caused the DepositPreauth amendment reference
  link to get overwritten so the table links to the feature page rather than the
  amendment description. Special cases this one link.

mDuo13 committed:
OfferCreate error cases: switch to table

mDuo13 committed:
DEX/Offers: edits per review

mDuo13 commented:
Applied the requested changes. Also, [46e43a2](https://github.com/XRPLF/xrpl-dev-portal/pull/1357/commits/46e43a2656ead7cdbd6a765205b98ca348cfe409) fixes some small issues with the Known Amendments page that came up in this process (see commit message for details).

amarantha-k reviewed (approved):
LGTM
